parser/src/main/java/com/linkedin/avroutil1/util/ConfigurableAvroSchemaComparator.java

```diff
@@ -39,13 +42,27 @@ public class ConfigurableAvroSchemaComparator {
   public static boolean equals(AvroSchema a, AvroSchema b, SchemaComparisonConfiguration config) {
     validateConfig(config);
     Set<SeenPair> seen = new HashSet<>(3);
+    List<AvroSchemaDifference> differences = new ArrayList<>();
     try {
-      return equals(a, b, config, seen);
+      equals(a, b, config, seen, differences);
+      return differences.isEmpty();
```

Comment:
Let's have `equals` call the old internal version of `private static boolean equals(a, b, config, seen)`. The old version failed fast, which is desirable for the `equals` use case.

In particular, Schema Registry uses this implementation to fail fast when checking for schema equivalence and using the changed implementation will be slower (since it finds all differences instead of failing fast)

Reply:
So my suggested implementation is to leave the old `private static boolean equals(a, b, config, seen)` as is and create a new internal method called `findDifferencesEquals` and have `findDifferences` call that helper method.

It would also be useful to add a line or 2 of documentation as to why we chose this implementation and added a bit of duplicated code :)

Reply:
Instead of re-adding `equals` method, added a flag`fastFail` to return on the first identified difference.